### PR TITLE
create product and review routes

### PR DESCRIPTION
**Pull request by eyezick**

### CODE REVIEW

File: server/products.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      .catch(next)
+  })
+
+  .get('/:id', (req, res, next) => {
+    res.send(req.product)
+  })
+
+
+  .get('/', (req, res, next) => {
+    Product.findAll().then(allProducts => res.send(allProducts)).catch(next)
```
Comment (ekotwick) [inline server/products.js]: to discuss: if the query returns no items, should we throw an error?

File: server/products.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    req.product.update(req.body)
+      .then(updatedProduct => {
+        res.send(updatedProduct)
+      })
+      .catch(next)
+  })
+
+
+  .delete('/:id', (req, res, next) => {
+    req.product.destroy().then(destroyedItem => console.log('destroyed',destroyedItem)).catch(next)
```
Comment (ekotwick) [inline server/products.js]: does `.destroy()` return the destroyed item? what does the function return?

**Review by ekotwick — CHANGES_REQUESTED** [review]
question about:
1) what `.destroy()` returns
2) error handling for `get('/')` route

File: server/products.js (hunk 2 above)
Comment (jkayen) [reply server/products.js]: i am pretty sure this does not return anything, so it should just be a function that takes no args and does `res.status(204).send(`deletion successful`)`

Comment (ekotwick) [reply server/products.js]: pretty sure that the http 204 response is a `no content` response.

so just do: `res.sendStatus(204)`

File: server/reviews.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const db = require('APP/db')
+const Review = db.model('review')
+
+module.exports = require('express').Router()
+
+  .param('id', (req, res, next, id) => {
+    Review.findbyId(id)
+      .then(foundReview => {
+        if (!foundReview) throw Error('omg error 404')
```
Comment (jkayen) [inline server/reviews.js:9]: USE SEMI-COLONS

File: server/reviews.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        req.review = foundReview
+        next()
+      })
+      .catch(next)
+  })
+
+  .get('/:id', (req, res, next) => {
+    res.send(req.review)
```
Comment (jkayen) [inline server/reviews.js]: use if statement to send a 404 and message that nothing was found at that id.

File: server/reviews.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      .then(createdReview => {
+        res.status(201).send(createdReview)
+      })
+      .catch(next)
+  })
+
+  .put('/:id', (req, res, next) => {
+    req.review.update(req.body)
+      .then(updatedReview => {
+        res.send(updatedReview)
```
Comment (jkayen) [inline server/reviews.js]: include an if statement per my comment above.

File: server/reviews.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      })
+      .catch(next)
+  })
+
+
+  .delete('/:id', (req, res, next) => {
+    req.review.destroy().then(destroyedItem => console.log('destroyed',destroyedItem)).catch(next)
```
Comment (jkayen) [inline server/reviews.js]: see comment above about delete

**Review by jkayen — CHANGES_REQUESTED** [review]
comment each route please. see comments for additional changes.

**Review by ekotwick — CHANGES_REQUESTED** [review]
comments

**Pull request merged**